pandas/core/nanops.py

```diff
@@ -474,8 +474,7 @@ def nanargmax(values, axis=None, skipna=True):
     """
     Returns -1 in the NA case
     """
-    values, mask, dtype, _ = _get_values(values, skipna, fill_value_typ='-inf',
-                                         isfinite=True)
```

Comment:
I think it better to put in the fix for inf, then in a followup fix the strings.